docs/content/stable/api/ysql/the-sql-language/statements/ddl_alter_table.md

```diff
@@ -368,6 +379,23 @@ Constraints marked as `INITIALLY IMMEDIATE` will be checked after every row with
 
 Constraints marked as `INITIALLY DEFERRED` will be checked at the end of the transaction.
 
+## Alter table operations that involve a table rewrite
+
+Most ALTER TABLE statements only involve a schema modification and complete quickly. However, certain specific ALTER TABLE statements require a new copy of the underlying table (and associated index tables, in some cases) to be made and can potentially take a long time, depending on the sizes of the tables and indexes involved. This is typically referred to as a "table rewrite". This behavior is [similar to PostgreSQL](https://www.crunchydata.com/blog/when-does-alter-table-require-a-rewrite), though the exact scenarios when a rewrite is triggered may differ between PostgreSQL and YugabyteDB.
+
+It is not safe to execute concurrent DML on the table during a table rewrite because the results of any concurrent DML are not guaranteed to be reflected in the copy of the table being made. This behavior is also similar to PostgreSQL, where a table lock typically excludes concurrent DML on the table during the rewrite.
```

Comment:
`This behavior is also similar to PostgreSQL, where a table lock typically excludes concurrent DML on the table during the rewrite.`

This indicates that PG has the consistency issues as well. May be slight bit rewrite. 

`PostgreSQL also has the same behaviour and safeguards the TableRewrite operation by  preventing the concurrent write operation by using the table lock.
`